Source/WebCore/platform/graphics/ImageBufferPixelFormat.h

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (C) 2024 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#include <wtf/Forward.h>
+
+namespace WebCore {
+enum class PixelFormat : uint8_t;
+enum class ImageBufferPixelFormat : uint8_t {
+    BGRX8,
+    BGRA8,
+#if HAVE(IOSURFACE_RGB10)
+    RGB10,
+    RGB10A8,
+#endif
+};
+
+WEBCORE_EXPORT static constexpr ImageBufferPixelFormat convertToImageBufferPixelFormat(PixelFormat format)
+{
+    switch (format) {
+    case PixelFormat::RGBA8:
+        ASSERT_NOT_REACHED();
+        return ImageBufferPixelFormat::BGRX8;
+    case PixelFormat::BGRX8:
+        return ImageBufferPixelFormat::BGRX8;
+    case PixelFormat::BGRA8:
+        return ImageBufferPixelFormat::BGRA8;
+#if HAVE(IOSURFACE_RGB10)
+    case PixelFormat::RGB10:
+        return ImageBufferPixelFormat::RGB10;
+    case PixelFormat::RGB10A8:
+        return ImageBufferPixelFormat::RGB10A8;
+#else
+    case PixelFormat::RGB10:
+    case PixelFormat::RGB10A8:
+        ASSERT_NOT_REACHED();
+        return ImageBufferPixelFormat::BGRX8;
+#endif
+    }
```

Comment:
```
   switch (format) {
    case PixelFormat::BGRX8:
        return ImageBufferPixelFormat::BGRX8;
    case PixelFormat::BGRA8:
        return ImageBufferPixelFormat::BGRA8;
#if HAVE(IOSURFACE_RGB10)
    case PixelFormat::RGB10:
        return ImageBufferPixelFormat::RGB10;
    case PixelFormat::RGB10A8:
        return ImageBufferPixelFormat::RGB10A8;
#endif
    default:
        break;
    }
```
This might be clearer and represents the opposite of the switch statement below.